aws/resource_aws_wafregional_web_acl.go

```diff
@@ -37,6 +42,44 @@ func resourceAwsWafRegionalWebAcl() *schema.Resource {
 					},
 				},
 			},
+			"logging_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
```

Comment:
General question: `MaxItems` implies the number of resources that can be defined within this TypeList, but since this resource happens to be a map of *scheme.Schema the resource has no bounds on the number of keys defined within it. Is that correct?

Reply:
Thats correct, the `MaxItems` on a `schema.TypeList` defines the number of configuration blocks allowed, not the number of nested attributes inside. `TypeList` and `MaxItems: 1` is a common pattern for resources to generate a single configuration block, e.g.

```hcl
resource "TYPE" "NAME" {
  logging_configuration {
    # ... nested attributes ...
  }
}
```

The following configuration will throw a validation error at plan time:

```hcl
resource "TYPE" "NAME" {
  logging_configuration {
    # ... nested attributes ...
  }

  logging_configuration {
    # ... nested attributes ...
  }
}
```

Reply:
:pray: thanks for explanation and example case, which I validated as well :wink: